demos/browser/app/meetingV2/meetingV2.ts

```diff
@@ -1907,45 +2004,60 @@ export class DemoMeetingApp
           this.appendNewSpeakerTranscriptDiv(segment, speakerToTranscriptSpanMap);
         } else {
           const transcriptSpan = speakerToTranscriptSpanMap.get(newSpeakerId);
-          transcriptSpan.innerText = transcriptSpan.innerText + '\u00a0' + segment.content;
+          transcriptSpan.innerText = transcriptSpan.innerText + '\u00a0';
+          transcriptSpan.appendChild(segment.contentSpan);
         }
       }
     }
   };
 
   populatePartialTranscriptSegmentsFromResult = (segments: TranscriptSegment[], result: TranscriptResult) => {
     let startTimeMs: number = null;
-    let content = '';
     let attendee: Attendee = null;
+    let contentSpan;
     for (const item of result.alternatives[0].items) {
+      const itemContentSpan = document.createElement('span') as HTMLSpanElement;
+      itemContentSpan.innerText = item.content;
+      itemContentSpan.classList.add('transcript-content');
+      // underline the word with red to show confidence level of predicted word being less than 0.3
+      if (item.hasOwnProperty('confidence') && !item.content.startsWith("[") && item.confidence < 0.3) {
+        itemContentSpan.classList.add('confidence-style');
+      }
+
+      // highlight the word in green to show the predicted word is a PII/PHI entity
+      if (this.transcriptEntitySet.size > 0 && this.transcriptEntitySet.has(item.content)) {
+        itemContentSpan.classList.add('entity-color');
+      }
+
       if (!startTimeMs) {
-        content = item.content;
+        contentSpan = document.createElement('span') as HTMLSpanElement;
+        contentSpan.appendChild(itemContentSpan);
         attendee = item.attendee;
         startTimeMs = item.startTimeMs;
       } else if (item.type === TranscriptItemType.PUNCTUATION) {
-        content = content + item.content;
+        contentSpan.appendChild(itemContentSpan);
         segments.push({
-          content: content,
+          contentSpan: contentSpan,
           attendee: attendee,
           startTimeMs: startTimeMs,
           endTimeMs: item.endTimeMs
         });
-        content = '';
         startTimeMs = null;
         attendee = null;
       } else {
         if (this.noWordSeparatorForTranscription) {
-          content = content + item.content;
+          contentSpan.appendChild(itemContentSpan);
         } else {
-          content = content + ' ' + item.content;
+          itemContentSpan.innerText = ' ' + itemContentSpan.innerText
```

Comment:
This will apply the entity/confidence styling to the space preceding the item content. Can we add two spans here, one for just the space without any classes on it and one for the item content?